modules/new_profile/jsx/NewProfileIndex.js

```diff
@@ -0,0 +1,260 @@
+   * for easy access by columnFormatter.
+   *
+   * @return {object}
+   */
+  fetchData() {
+    return fetch(this.props.dataURL, {credentials: 'same-origin'})
+      .then((resp) => resp.json())
+      .then((data) => this.setState({configData: data}))
+      .catch((error) => {
+        this.setState({error: true});
+        console.error(error);
+      });
+  }
+
+    validate() {
+        const data = this.state.formData;
+        let isError = false;
+        if (data.dateTaken !== data.dateTakenConfirm) {
+            isError = true;
+        }
+        let decError = false;
+        if (this.state.configData['edc'] === 'true'
+            && data.edcDateTaken !== data.edcDateTakenConfirm
+        ) {
+            decError = true;
+        }
+        return isError || decError;
+    }
+
+
+    /**
+   * Handles form submission
+   *
+   * @param {event} e - Form submition event
+   */
+    handleSubmit(e) {
+        e.preventDefault();
+        const err = this.validate();
+        this.setState(
+            {
+                errMessage: '',
+                }
+        );
+        if (err) {
+            this.setState(
+                {
+                    errMessage: 'Date of birth fields must match',
+                    isCreated: false,
+                }
+            );
+        } else {
+    let formData = this.state.formData;
+    let formObject = new FormData();
+    for (let key in formData) {
+      if (formData[key] !== '') {
+        formObject.append(key, formData[key]);
+      }
+    }
+/* eslint-disable no-console */
+   fetch(this.props.submitURL, {
+      method: 'POST',
+      cache: 'no-cache',
+      credentials: 'same-origin',
+      body: formObject,
+    })
+    .then((resp) => resp.json())
+    .then((data) => {
+     this.setState({newData: data});
+     this.setState({isCreated: true});
+    });
+                }
+    }
+/* eslint-enable no-console */
+
+    /**
+   * Set the form data based on state values of child elements/componenets
+   *
+   * @param {string} formElement - name of the selected element
+   * @param {string} value - selected value for corresponding form element
+   */
+    setFormData(formElement, value) {
+        let formData = JSON.parse(JSON.stringify(this.state.formData));
+        formData[formElement] = value;
+
+        this.setState(
+            {
+                formData: formData,
+            }
+        );
+    }
+
+     render() {
+            let profile = null;
+            let edc = null;
+            let project = null;
+            let pscid = null;
+            if (this.state.configData['useProject'] === 'true') {
+                project =
+                <div>
+                 <SelectElement
+                   name ="project"
+                   label ="Project"
+                   options ={this.state.configData.project}
+                   onUserInput ={this.setFormData}
+                   ref ="project"
```

Comment:
I cna't remember what the problem with `ref=` is.. is this kosher? @davidblader @HenriRabalais ?

Reply:
No string refs since they're legacy and likely to be deprecated. https://reactjs.org/docs/refs-and-the-dom.html#legacy-api-string-refs